theano/gpuarray/basic_ops.py

```diff
@@ -297,6 +297,9 @@ def get_params(self, node):
         # Default implementation, suitable for most sub-classes.
         # To be necessarly overridden if a subclass uses a ParamsType
         # (see GpuCumOp, GpuAdvancedIncSubtensor1_dev20 for examples).
+        assert (self.params_type is gpu_context_type and
+                node.inputs and
+                isinstance(node.inputs[0].type, GpuArrayType))
```

Comment:
In fact, every GpuKernelBase sub-class that does not get context from first input currently overrides `get_params()`, and I have also overridden it in the Ops that I have modified.

**NB:**
My initial goal was to factorize get_params() because:
 * All GpuKernelBase sub-classes previously defined get_params()
 * About half of all these sub-classes defined get_params() with the same manner: `return node.inputs[0].type.context`.
So, adding this default implementation to GpuKernelBase helped remove many lines of codes in whole code of sub-classes, while keeping get_params() overridden in sub-classes that get context differently. See commit https://github.com/Theano/Theano/pull/5866/commits/c9efa82453dbe28a541fed2791aac92324744054 for the factorization.

I could revert it if necessary !